### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by arisslin**
- add favicon
- create custom hooks
- refactor styled components
- refactor utils

### CODE REVIEW

File: src/common/utils.js [hunk 1]
```diff
@@ -1,19 +1,37 @@
...
     element.current.focus()
   })
 }
 
+export function getKeyByName(pads, name) {
+  return pads.reduce((acc, cur) => {
+    if (name === toLowerNoWhiteSpace(cur.name)) {
+      acc = acc + cur.key
+    }
+    return acc
```
Comment (jamarob) [inline src/common/utils.js:12]: Maybe use a ternary.

File: src/common/utils.js [hunk 2]
```diff
@@ -1,19 +1,37 @@
...
+  }, '')
+}
+
+export function getPadNameByKey(pads, key) {
+  return pads.reduce((acc, curr) => {
+    if (key === curr.key) {
+      acc = acc + curr.name
+    }
+    return acc
+  }, '')
```
Comment (jamarob) [inline src/common/utils.js:22]: Maybe use a ternary.

File: src/hooks/usePads.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  }
+
+  function handleKeyUp(event) {
+    const key = event.key
+    setPadIsTriggered(key, false)
+  }
+
+  function setPadIsTriggered(key, triggerd) {
+    const index = pads.findIndex(pad => pad.key === key)
+    if (index > -1) {
```
Comment (jamarob) [inline src/hooks/usePads.js:33]: Is there a real chance, that the pad is not found?

File: src/hooks/useVolume.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import { useState } from 'react'
+
+export default function useVolume(player) {
+  const [volume, setVolume] = useState(player.getMasterVolume())
+
+  function adjustVolume(event) {
+    const targetValue = Number(event.target.value)
+    const volumeInDB = targetValue > 6 ? 6 : targetValue
```
Comment (jamarob) [inline src/hooks/useVolume.js:8]: Maybe introduce a const `MAX_VOLUME` for the magic number?

File: src/hooks/useVolume.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import { useState } from 'react'
+
+export default function useVolume(player) {
+  const [volume, setVolume] = useState(player.getMasterVolume())
+
+  function adjustVolume(event) {
```
Comment (jamarob) [inline src/hooks/useVolume.js:6]: Think about passing in a number and decoupling this hook from events?

File: package.json [hunk 6]
```diff
@@ -16,6 +13,10 @@
     "tone": "13.8.25"
   },
   "scripts": {
+    "@testing-library/jest-dom": "4.2.4",
+    "@testing-library/react": "9.4.1",
+    "@testing-library/user-event": "7.2.1",
+    "prop-types": "15.7.2",
```
Comment (jamarob) [inline package.json:19]: I think these need to go into `devDependencies`!

File: package.json [hunk 7]
```diff
@@ -16,6 +13,10 @@
...
     "predeploy": "npm run build",
```
Comment (jamarob) [inline package.json:20]: This just calls the default build script.

File: src/hooks/usePads.js (hunk 3 above)
Comment (arisslin) [reply src/hooks/usePads.js:33]: Yes, wenn the key is not matched

**Pull request merged**